apps/cms.c

```diff
@@ -913,12 +918,24 @@ int cms_main(int argc, char **argv)
             goto end;
     }
 
-    if ((flags & CMS_BINARY) == 0)
-        warn_binary(infile);
-    in = bio_open_default(infile, 'r',
-                          binary_files ? FORMAT_BINARY : informat);
-    if (in == NULL)
-        goto end;
+    if (digesthex != NULL) {
+        if (infile != NULL ||
+            operation != SMIME_SIGN ||
```

Comment:
Our coding style puts the `||` at the start of the following line...

Reply:
Done.